runtime/gc_include/ObjectAccessBarrierAPI.hpp

```diff
@@ -70,14 +70,15 @@ class MM_ObjectAccessBarrierAPI
 	 * Create an instance.
 	 */
 	MM_ObjectAccessBarrierAPI(J9VMThread *currentThread)
-		: _writeBarrierType(currentThread->javaVM->gcWriteBarrierType)
+		:
+#if defined(OMR_GC_COMPRESSED_POINTERS) && defined(OMR_GC_FULL_POINTERS)
+		_compressObjectReferences(J9VMTHREAD_COMPRESS_OBJECT_REFERENCES(currentThread))
+#endif /* defined(OMR_GC_COMPRESSED_POINTERS) && defined(OMR_GC_FULL_POINTERS) */
+		, _writeBarrierType(currentThread->javaVM->gcWriteBarrierType)
```

Comment:
This won't compile when `!(defined(OMR_GC_COMPRESSED_POINTERS) && defined(OMR_GC_FULL_POINTERS))`.

Reply:
I'm going to move the `protected` section so that the field doesn't come first. Reordering should be okay to do per https://github.com/eclipse/omr/pull/5758#discussion_r560411683